### PR TITLE
Convert attach data to UTF16 before consuming it.

### PR DESCRIPTION
**Pull request by jander-msft**
- Convert attach data to UTF16 before consuming it.
- Fix configuration XML memory leak.

### CODE REVIEW

**Comment by WilliamXieMSFT** [comment]
>     const size_t WCharSizeInBytes = sizeof(WCHAR) / sizeof(BYTE);

Not used anymore

---
Refers to: src/InstrumentationEngine/ProfilerManager.h:25 in dc539b5. [](commit_id = dc539b53f245250964243fa7cf983b3648e9209f, deletion_comment = False)

File: src/InstrumentationEngine/ProfilerManager.h [hunk 1]
```diff
@@ -82,7 +82,7 @@ namespace MicrosoftInstrumentationEngine
         vector<CComPtr<CConfigurationSource>> m_configSources;
 
         // The configuration xml for profiler attach.
-        LPCWSTR m_wszConfigXml;
```
Comment (WilliamXieMSFT) [inline src/InstrumentationEngine/ProfilerManager.h]: >LPCWSTR [](start = 8, length = 7)

Was this what caused the memory leak?

Comment (jander-msft) [reply src/InstrumentationEngine/ProfilerManager.h]: Yes

**Pull request merged**